frame/proxy/src/lib.rs

```diff
@@ -60,14 +60,12 @@ pub trait Trait: frame_system::Trait {
 
 	/// The overarching call type.
 	type Call: Parameter + Dispatchable<Origin=Self::Origin, PostInfo=PostDispatchInfo>
```

Comment:
This is outer call, the one from `add_filter` is also outer call, and we need `IsType`  to only convert from outer call... to outer call again. have you considered moving the bounds `type Call` in system, and just use that as the argument of this module. 

In other words, I am suggesting that this module does not need to have `type Call` and can just use the `type Call` from system in all places, potentially removing `IsType`.

Reply:
the Filter argument is `frame_system::Trait::Call` but proxy pallet want to bound  `IsSubType<ProxyModule<..>,..>` on it.
We can't add this bound in frame_system because frame_system is not aware of the other modules.
The only way to add additional bound on a inherited trait (`pub Trait trait: frame_system::Trait`) is to use a where clause: `where <Self as frame_system::Trait>::Call: IsSubType<ProxyModule<Self>, Self>` this where clause works but has to be written everytime we make this use.

Reply:
okay makes sense to just use `IsType`.